docs/implementation.html

```diff
@@ -350,12 +350,28 @@
 /consumers/[group_id]/owners/[topic]/[partition_id] --> consumer_node_id (ephemeral node)
 </pre>
 
+<h4><a id="impl_clusterid" href="#impl_clusterid">Cluster Id</a></h4>
+
+<p>
+    The cluster id is an unique and immutable identifier assigned to a Kafka cluster. The cluster id is 22 characters long and the allowed characters is defined by the regular expression [a-zA-Z0-9_\-]+, which corresponds to the characters used by the URL-safe Base64 variant. It is auto-generated when the cluster is started for the first time.
```

Comment:
"a unique" (yes, this is a bit weird: https://en.oxforddictionaries.com/definition/unique)

Also, I think we need to clearly distinguish between specified behaviour and implementation details. The specified behaviour is that it's a unique and immutable string of no more than 22 characters where the allowed character set is the same as the URL-safe base64 variant with no padding (the regex is useful too). These details help people if they want to store this in a DB or use it in places where the character set is restricted.

Then, the implementation detail is that we generate 22 characters today. But, we could potentially reduce this in the future and users should not depend on it being exactly 22 characters.

Does that make sense?

Also, we could perhaps say:

"Conceptually, it is auto-generated when a cluster is started for the first time. Implementation-wise, it is generated the first time a broker with version 0.10.1 or later connects to ZooKeeper for the first time".